### DISCUSSION TITLE
Track date & IP of user-verification (on user registration)

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#22784] [Feature Requests]
### Summary

Since directus v10.11 there's a specific functionality for user-registration and verification
In order to track user verification better, I suggest to store the verification date as well as IP together with the user.

### Motivation

Keep a documentation when / from where a user has been verified. Saving the timestamp can be used as a documentation, storing the IP can be used to analyze / track spam registrations as well as documentation, too.

### Detailed Design

- Extend the user-table with two new fields: 
  - `verified_at` (datetime)
  - `verification_ip` (string)
- Update the Verification service to store the data:

https://github.com/directus/directus/blob/53b3c1fa0049daa21b123ce98ce3c8b6e482fd7c/api/src/services/users.ts#L596-L613

### Must Have:
- Tracking verification date, to store verification proof

### Should Have:
- IP from verification

### Drawbacks

Data protection could be a thing, but as the directus_activities already track the IP it wouldn't be a change

### Alternatives

/

### Adoption Strategy

see above

**Comment by rijkvanzanten** [comment]
`directus_activity` already stores the IP and timestamps of the changes to those user records. What would be the benefit of also storing that information on the user directly?

**Reply by Dominic-Marcelino** [reply]
For me it's:
- from my pov it's an own data-attribute and not just an activity
- currently it's safe to purge `directus_activity` (e.g when it grows) without loosing any data. Seeing the verification-meta as it own data this would be a data loss (related to #17894)
- Personal preference: from my pov a user-verification should be easily fetchable with a user, not requiring filtering another (potentially huge) itemset

**Comment by hanneskuettner** [comment]
I'm wondering if this feature should be implemented in the core itself or if it makes more sense to surface a separate `users.verified` action, that can be used to do whatever you please with the provided information, including storing fields from the `accountability` to the user 🤔  Seems more flexible IMO

**Comment by Dominic-Marcelino** [comment]
> surface a separate `users.verified` action, that can be used to do whatever you please with the provided information

Would be an alternative, but in that case also possible by using the normal users actions as it's just a status change 🤔

**Reply by hanneskuettner** [reply]
The problem I can see there is that the `users.update` action does not know about the before state of the user, so the transition from `invited` to `active` is not traceable, and setting the verified fields whenever the status is `active` does not seems correct. One could definitely do something with field presets, checking the current values of the verified fields and all of that. 

Having a dedicated action does seem a lot more convenient.

**Reply by Dominic-Marcelino** [reply]
Indeed, true! I'm open to write & publish an extension once a hook (or other solution) becomes available 😄

**Comment by Dominic-Marcelino** [comment]
Now that the `RETENTION_ENABLED` configuration arrived I think it's even more important to have the option to easily track the verification as the `activity` entry could be deleted. Are you open to convert this into an issue / PR for a `users.verified` action?
